src/container.cpp

```diff
@@ -736,27 +736,27 @@ void Container::startDecaying()
 {
 	Item::startDecaying();
 
-	for (Item* item : itemlist) {
+	for (auto item : itemList) {
 		item->startDecaying();
 	}
 }
 
 ContainerIterator Container::iterator() const
 {
 	ContainerIterator cit;
-	if (!itemlist.empty()) {
-		cit.over.push_back(this);
-		cit.cur = itemlist.begin();
+	if (!itemList.empty()) {
+		cit.over.push_back(getContainer());
+		cit.cur = itemList.begin();
 	}
 	return cit;
 }
 
-Item* ContainerIterator::operator*() { return *cur; }
+std::shared_ptr<Item> ContainerIterator::operator*() { return *cur; }
 
 void ContainerIterator::advance()
 {
-	if (Item* i = *cur) {
-		if (Container* c = i->getContainer()) {
+	if (auto i = cur->get()) {
+		if (auto c = i->getContainer()) {
```

Comment:
Could this be `const auto`
```suggestion
		if (const auto c = i->getContainer()) {
```